### PR TITLE
CI Visibility - Test configurations guide

### PR DESCRIPTION
**Pull request by nachoBonafonte**
### What does this PR do?

It adds a guide about how test configurations work and how to add custom configurations

### Motivation

We want to explain our users how test configurations work

https://docs-staging.datadoghq.com/nachoBonafonte/CI-Visibility-Configurations-Guide/continuous_integration/guides/test_configurations

### Additional Notes

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Check images for PII
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/continuous_integration/guides/test_configurations.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+In CI Visibility, a test with multiple configurations is treated as multiple tests; a separate test for each configuration. In the case where one of the configurations fails but the others pass, only that specific test and configuration combination will be marked as failed.
+
+For example, if you're testing a single commit and you have a python test that runs against three different python versions, if the test fails for one of those versions, that specific test will be marked as failed, while the other versions will be marked as passed. However, if you retry the tests against the same commit and now the test for all three python versions pass, the test with the version that previously failed will now be marked as both passed and flaky, while the other two versions will remain passed, with no flakiness detected.
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:14]:
```suggestion
For example, if you're testing a single commit and you have a Python test that runs against three different Python versions, if the test fails for one of those versions, that specific test will be marked as failed, while the other versions will be marked as passed. However, if you retry the tests against the same commit and now the test for all three Python versions pass, the test with the version that previously failed will now be marked as both passed and flaky, while the other two versions will remain passed, with no flakiness detected.
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+---
+title: Test Configurations
+kind: guide
+---
+
+{{< site-region region="gov" >}}
+<div class="alert alert-warning">CI Visibility is not available in the selected site ({{< region-param key="dd_site_name" >}}) at this time.</div>
+{{< /site-region >}}
+
+Tests evaluate the behavior of code for a set of given conditions. Some of those conditions are related to the runtime and settings of the system where the tests are run. The same code run in different conditions can have a different behavior, so developers usually configure their tests in order to run in different conditions and validate that the behavior is the expected for all of them.
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:10]: @nachoBonafonte I'm realizing that here we talk about "conditions" and later we talk about "configurations" without describing what's the relationship between them. What about doing something like this (pending editorial review of course):

```suggestion
Tests evaluate the behavior of code for a set of given conditions. Some of those conditions are related to the environment where the tests are run, such as the operating system or the runtime used. The same code executed different sets of conditions can behave differently, so developers usually configure their tests to run in different sets of conditions and validate that the behavior is the expected for all of them. A specific set of conditions is what we call a "configuration".
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+For example, if you're testing a single commit and you have a python test that runs against three different python versions, if the test fails for one of those versions, that specific test will be marked as failed, while the other versions will be marked as passed. However, if you retry the tests against the same commit and now the test for all three python versions pass, the test with the version that previously failed will now be marked as both passed and flaky, while the other two versions will remain passed, with no flakiness detected.
+
+### Default configurations
+
+When you are running your tests with CI visibility, the library automatically detects and reports information about the conditions that tests are being run in the test tags. For example, if the tests were running in a windows or linux environment, or the architecture of the platform running the tests (arm64, x86_64) those values are reported on the test tags. These values related to configurations are shown in the interface when a test fails or shows flakiness (only those configurations that make a difference are shown):
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:18]:
```suggestion
When you are running your tests with CI Visibility, the library automatically detects and reports information about the environment where tests are being run as test tags. For example, the operating system name (e.g. Windows, Linux) and the architecture of the platform (e.g. arm64, x86_64) are reported as tags on each test. These values are shown in the commit and branch overview pages when a test fails or is flaky for a specific configuration but not others. Note that only the configuration tags that change between different test runs are shown on the UI.
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+{{< img src="ci/test_configurations_in_errors.png" alt="Configuration for test failures" style="width:100%;">}}
+
+The tags that are currently being used to automatically identify the configuration for the tests are the following tags (some are platform dependant and don't exist in all languages):
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:22]:
```suggestion
These are the tags that are automatically collected to identify test configurations. Note that some only apply to specific platforms:
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+* `device.name` - Name of the device
+* `ui.appearance` - User Interface style
+* `ui.orientation` - Orientation the UI is run
+* `ui.localization` - Language of the application
+
+### Custom configurations
+
+There are some configurations that cannot be directly identified and reported automatically because they can depend on environment variables, on test run arguments, or other approaches that developers use. For those cases the developer must provide the configuration to the library so CI Visibility can properly identify them.
+
+ The user must define those tags as part of the DD_TAGS environment variable using the test.configuration prefix, e.g:
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:41]:
```suggestion
These tags must be defined as part of the `DD_TAGS` environment variable using the `test.configuration` prefix, for example:
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+DD_TAGS=test.configuration.disk:slow,test.configuration.memory:low
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:43]:
```suggestion
{{< code-block lang="bash" >}}
DD_TAGS=test.configuration.disk:slow,test.configuration.memory:low
{{< /code-block >}}
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+All the entries under the test.configuration prefix will be used as different configuration settings for the tests that include them.
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:45]:
```suggestion
All tags with the `test.configuration` prefix will be used as configuration tags, in addition to the automatically collected ones.
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+
+Note: If the developer wants to filter using these configurations, facets must be explicitly created for these tags. [Facets Documentation][1]
+
+
+[1]: tracing/trace_explorer/facets/#managing-facets
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:50]:
```suggestion
[1]: tracing/trace_explorer/facets/#creating-facets
```

File: content/en/continuous_integration/guides/test_configurations.md [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+There are some configurations that cannot be directly identified and reported automatically because they can depend on environment variables, on test run arguments, or other approaches that developers use. For those cases the developer must provide the configuration to the library so CI Visibility can properly identify them.
+
+ The user must define those tags as part of the DD_TAGS environment variable using the test.configuration prefix, e.g:
+
+DD_TAGS=test.configuration.disk:slow,test.configuration.memory:low
+
+All the entries under the test.configuration prefix will be used as different configuration settings for the tests that include them.
+
+Note: If the developer wants to filter using these configurations, facets must be explicitly created for these tags. [Facets Documentation][1]
```
Comment (fermayo) [inline content/en/continuous_integration/guides/test_configurations.md:47]:
```suggestion
Note that in order to filter using these configurations tags, [facets must be explicitly created for these tags][1].
```

**Comment by alai97** [comment]
Thanks for reviewing @fermayo! I created an editorial review card for the Docs team: DOCS-3907.

**Comment by fermayo** [comment]
Thanks @alai97 !

**Review by kayayarai — APPROVED** [review]
Hi! I made some edits for style and readability, and added some links for further reading. Please check I didn't mess anything up. If you're happy with it, let me know and I will merge

**Comment by fermayo** [comment]
@kayayarai LGTM, but we are still waiting for this functionality to be released. We'll be removing the `do-not-merge` tag and ping you once it's ready to be merged. Thanks!

**Comment by nachoBonafonte** [comment]
Hi @kayayarai, the functionality has been released, we can merge this PR. Thanks!

**Pull request merged**